### DISCUSSION TITLE
Error when I try to swap texture froma script.js

### DISCUSSION THREAD

**Discussion by digityze2023** [google/model-viewer#4303] [Q&A] [answered]
Dears,
starting from https://modelviewer.dev/examples/scenegraph/index.html#swapTexturesExample when I put the script code insieme a script.js file the swap texture does not work.

When I use the code in the pase as it is there is no issue.

Have you an idea? 

I like to have a clear html page and I don't want to put script explicit code inside it.

**Comment by digityze2023** [comment]
I suppose there is something not clear to me on the js coding..

`
const modelViewerPhoto = document.getElementById("model");

async function downloadPosterToBlob() {
  const blob = await modelViewerPhoto.toBlob({ idealAspect: false });
  const url = URL.createObjectURL(blob);
  const a = document.createElement("b");
  a.href = url;
  a.download = "screenshot-model.png";
  a.click();
  URL.revokeObjectURL(url);
}

function esporta() {
    const glTF = await modelViewerPhoto.exportScene();
    const file = new File([glTF], "export.glb");
    const link = document.createElement("a");
    link.download =file.name;
    link.href = URL.createObjectURL(file);
    link.click();
  }`
  
  Why it is not working?
  
  I think that when I will fix it I will fix the other swap texture issue

**Reply by elalish** [reply, marked as answer]
Hard to say without seeing the page, but I would bet it's an async issue. Make sure if you are trying to read or change the GLB in any way that it has actually loaded first: put all that code inside of a `load` event listener.
